Automattic/jetpack
# Implement `setOptOut` cookie command

**NOTE:** Please consider merging https://github.com/Automattic/jetpack/pull/9156 instead of this.

---

### PR Objectives

- Implement the upcoming `setOptOut` command; i.e., whenever the new [privacy option](https://github.com/Automattic/jetpack/pull/8993) is changed, this PR will set or unset the opt-out cookie using `window._tkq.push( [ 'setOptOut', true|false ] )`

- Make `Jetpack_Tracks_Client` in PHP aware of this cookie too. If the opt-out cookie exists, do not record server-side events either.

### How to test:
- ~~Depends on #8993 and #9003~~ These are merged already.
- In a browser console, run: `localStorage.setItem( 'debug', 'dops:analytics' );`
- Open the new Privacy setting described and implemented by #8993
- Toggle the privacy option on/off and observe browser console
- Expect to see `Pushing setOptOut: true|false`

---

### Blocker

This will not actually set the cookie, it merely logs what would occur if `setOptOut` existed. Therefore, this PR cannot be tested any further at this time. The work in gh-analytics-38 (`setOptOut` command) needs to exist in `w.js` as well, which I am investigating.

### ~~Unresolved Questions~~

- ~~Will the final option key for opt-outs in Jetpack remain `jetpack_event_tracking`? This question has been raised in: p7kMJG-4rC-p2 and is pending further discussion.~~ **Yes**
- ~~Should `setOptOut` disable Google Analytics in addition to Tracks?~~ **No**
